plugins/modem-manager/fu-mm-device.c

```diff
@@ -285,15 +297,32 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
 	for (guint i = 0; device_ids[i] != NULL; i++)
 		fu_device_add_instance_id (device, device_ids[i]);
 	if (fu_device_get_vendor_ids (device) == NULL) {
-		g_autofree gchar *path = g_build_filename (device_sysfs_path, "idVendor", NULL);
-		g_autofree gchar *value = NULL;
+		g_autofree gchar *path = NULL;
+		g_autofree gchar *value_str = NULL;
 		g_autoptr(GError) error_local = NULL;
 
-		if (!g_file_get_contents (path, &value, NULL, &error_local)) {
+		if (g_strcmp0 (device_bus, "USB") == 0)
+			path = g_build_filename (device_sysfs_path, "idVendor", NULL);
+		else if (g_strcmp0 (device_bus, "PCI") == 0)
+			path = g_build_filename (device_sysfs_path, "vendor", NULL);
+
+		if (path == NULL) {
+			g_warning ("failed to set vendor ID: unsupported bus: %s", device_bus);
+		} else if (!g_file_get_contents (path, &value_str, NULL, &error_local)) {
 			g_warning ("failed to set vendor ID: %s", error_local->message);
 		} else {
-			g_autofree gchar *vendor_id = g_strdup_printf ("USB:0x%s", g_strchomp (value));
-			fu_device_add_vendor_id (device, vendor_id);
+			guint64 value_int;
+
+			/* note: the string value may be prefixed with '0x' (e.g. when reading
+			 * the PCI 'vendor' attribute, or not prefixed with anything, as in the
+			 * USB 'idVendor' attribute. */
+			value_int = g_ascii_strtoull (value_str, NULL, 16);
```

Comment:
does fu_common_strtoull help you here?

Reply:
No, it doesn't, because the `value_str` is always in base16, so we shouldn't guess the base. `fu_common_strtoull()` would read the string as base 10 if it doesn't have the `0x` prefix.